src/daft-scan/src/lib.rs

```diff
@@ -755,6 +739,14 @@ impl PartitionField {
             }),
         }
     }
+
+    pub fn clone_field(&self) -> Field {
+        self.field.clone()
+    }
+
+    pub fn clone_source_field(&self) -> Option<Field> {
+        self.source_field.clone()
```

Comment:
I feel like these two are unncessary, it make be better just to do `field.clone` so the caller knows that the field is getting cloned

Reply:
Thanks for the suggestion, turns out we can remove this second clone method without any impact